media.go

```diff
@@ -116,6 +117,15 @@ type producerConsumer[T any, U any] struct {
 // for error handling logic based on consecutively retrieved errors).
 type ErrorHandler func(ctx context.Context, mediaErr error)
 
+func PropertiesFromMediaSource[T, U any](src MediaSource[T]) []prop.Media {
+	if asMedia, ok := src.(*mediaSource[T, U]); ok {
+		if asMedia.driver != nil {
+			return asMedia.driver.Properties()
+		}
+	}
+	return make([]prop.Media, 0)
```

Comment:
just nil is sufficient in go

Reply:
You won't be able to take the len of `nil` though right? So the caller would have to check for nil _and_ the len as opposed to just getting the length.

Reply:
You can take the Len of nil

Reply:
Done!